infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/LogReplicationFSM.java

```diff
@@ -251,7 +252,7 @@ public LogReplicationFSM(DataSender dataSender,ExecutorService workers, LogRepli
      */
     @VisibleForTesting
     public LogReplicationFSM(SnapshotReader snapshotReader, DataSender dataSender,
-                             LogEntryReader logEntryReader, ExecutorService workers, LogReplicationAckReader ackReader,
+                             LogEntryReader logEntryReader, ScheduledExecutorService workers, LogReplicationAckReader ackReader,
```

Comment:
With Shama's PR [3815](https://github.com/CorfuDB/CorfuDB/pull/3815), there is a way to add an event with a delay and this change will not be required.  Just FYI.